src/quote/swap/adapters/uniswap/index.ts

```diff
@@ -0,0 +1,136 @@
+    const { chainId, inputAmount, outputAmount, slippage } = request
+
+    const inputToken = changeToWethIfNecessary(request.inputToken, chainId)
+    const outputToken = changeToWethIfNecessary(request.outputToken, chainId)
+
+    if (
+      isSameAddress(inputToken, EthAddress) ||
+      isSameAddress(outputToken, EthAddress)
+    ) {
+      // FIXME: remove for production, just for runnint tests and catching any of these cases
+      throw new Error('Error - using ETH instead of WETH')
+    }
+
+    if (!inputAmount && !outputAmount) {
+      throw new Error('Error - either input or output amount must be set')
+    }
+
+    const weth = getTokenData('WETH', chainId)
+    const inputTokenData = getTokenDataByAddress(inputToken, chainId)
+    const outputTokenData = getTokenDataByAddress(outputToken, chainId)
+
+    if (!inputTokenData || !outputTokenData || !weth) {
+      throw new Error('Error - either input or output token is invalid')
+    }
+
+    try {
+      const isExactOutput = outputAmount !== undefined
+
+      const inputTokenCurrency = new Token(
+        chainId,
+        inputToken,
+        inputTokenData.decimals
+      )
+      const outputTokenCurrency = new Token(
+        chainId,
+        outputToken,
+        outputTokenData.decimals
+      )
+
+      const router = new AlphaRouter({
+        chainId,
+        provider: getRpcProvider(this.rpcUrl),
+      })
+
+      let swapRoute: SwapRoute | null = null
+      if (isExactOutput) {
+        swapRoute = await router.route(
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          CurrencyAmount.fromRawAmount(outputTokenCurrency, outputAmount!),
+          inputTokenCurrency,
+          TradeType.EXACT_OUTPUT
+        )
+      } else {
+        swapRoute = await router.route(
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          CurrencyAmount.fromRawAmount(inputTokenCurrency, inputAmount!),
+          outputTokenCurrency,
+          TradeType.EXACT_INPUT
+        )
+      }
+
+      if (!swapRoute) return null
+
+      const trade = swapRoute.trade
+      if (!trade) return null
+
+      const swap = trade.swaps[0]
+      const quotedAmount = isExactOutput
+        ? swap.inputAmount.quotient.toString()
+        : swap.outputAmount.quotient.toString()
+
+      const route = swap.route
+      const path: string[] = route.path.map((token) => token.address)
+
+      const isV3 = route.protocol === 'V3'
+      let fees: number[] = []
+      if (isV3) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        fees = route.pools.map((pool: any) => pool.fee)
+      } else {
+        fees = route.pools.map(() => 3000)
+      }
+
+      return {
+        chainId,
+        inputToken,
+        outputToken,
+        inputAmount: inputAmount ?? quotedAmount,
+        outputAmount: outputAmount ?? quotedAmount,
+        callData: '0x', // TOOD:
```

Comment:
@ckoopmann I would assume this is the call data for the 0x contract right? Or would it work to send a Uniswap encoded tx call data here?

This should be only relevant for the FlashMintZeroEx contract, correct? https://github.com/IndexCoop/flash-mint-sdk/blob/main/src/quote/flashmint/zeroEx/provider.ts#L91